cmd/norduser/main.go

```diff
@@ -1,5 +1,42 @@
 package main
 
+import (
+	"log"
+	"os"
+	"strconv"
+
+	"google.golang.org/grpc"
+
+	"github.com/NordSecurity/nordvpn-linux/internal"
+	"github.com/NordSecurity/nordvpn-linux/norduser"
+	"github.com/NordSecurity/nordvpn-linux/norduser/pb"
+)
+
+var ConnURL = internal.GetNorduserdSocket(os.Geteuid())
+
 func main() {
-	<-make(chan struct{})
+	var listenerFunction = internal.SystemDListener
+	if os.Getenv(internal.ListenPID) != strconv.Itoa(os.Getpid()) {
+		listenerFunction = internal.ManualListener(ConnURL, internal.PermUserRWX)
+	}
+
+	listener, err := listenerFunction()
+	if err != nil {
+		log.Fatalf("Error on listening to UNIX domain socket: %s\n", err)
+	}
+
+	server := norduser.NewServer()
+
+	grpcServer := grpc.NewServer()
+	pb.RegisterNorduserServer(grpcServer, server)
+
+	go func() {
+		if err := grpcServer.Serve(listener); err != nil {
```

Comment:
If this fails, does it make sense for the app to continue running?

Reply:
`log.Fatalln` will terminate the program.